packages/flow-core/src/components/f-document-viewer/f-document-viewer.ts

```diff
@@ -0,0 +1,542 @@
+					const rect = contentRefs[i].getBoundingClientRect();
+					const distance = Math.abs(rect.top);
+					if (distance < minDistance) {
+						minDistance = distance;
+						activeSectionIndex = i;
+					}
+				}
+
+				sidebarRefs.forEach((sidebar, i) => {
+					if (i === activeSectionIndex) {
+						sidebar.selected = "background";
+						setTimeout(() => {
+							sidebar.scrollIntoView({
+								behavior: "auto",
+								block: "start"
+							});
+						}, 1000);
+					} else {
+						sidebar.selected = "none";
+					}
+				});
+			};
+
+			this.fDocViewerScrollContent?.addEventListener("scroll", handleScroll);
+		}
+	}
+
+	/**
+	 *
+	 * @param obj complete json object for doc creation
+	 * @param level until which level of nesting the data must be displayed
+	 * @returns the filtered data to a particular level
+	 */
+	filterObjectByLevel(obj: FDocViewerContent, level: number) {
+		const filteredObj = cloneDeep(obj);
+		function traverse(obj: FDocViewerContent, currentLevel: number) {
+			if (currentLevel === level) {
+				for (const key in obj) {
+					if (typeof obj[key] === "object" && obj[key] !== null) {
+						obj[key] = (obj[key] as FDocumentStatement).title;
+						delete (obj[key] as FDocumentStatement).data;
+					}
+				}
+			} else {
+				for (const key in obj) {
+					if (typeof obj[key] === "object" && obj[key] !== null) {
+						traverse((obj[key] as FDocumentStatement).data ?? {}, currentLevel + 1);
+					}
+				}
+			}
+		}
+		traverse(filteredObj, 0);
+		return filteredObj;
+	}
+
+	/**
+	 *
+	 * @param level level to be selected
+	 */
+	handleSelection(e: CustomEvent) {
+		const str = e.detail.value as string;
+		const level = Number(str.charAt(str.length - 1));
+		this.indexingLevelSelected = level;
+		this.filteredContents = this.filterObjectByLevel(this.content, level);
+	}
+
+	/**
+	 *
+	 * @param e value to be searched
+	 */
+	handleSearch(e: CustomEvent) {
+		this.searchValue = e.detail.value;
+		this.requestUpdate();
+	}
+
+	toggleLeftColumn() {
+		const previousState = this.leftColumn.getAttribute("data-column-open") === "true";
+
+		if (!previousState) {
+			this.leftColumnWrapper.width = "300px";
+			this.leftColumnWrapper.border = "small solid secondary right";
+			this.leftColumnWrapper.setAttribute("class", "jumplinks-expanded");
+		} else {
+			this.leftColumnWrapper.width = "0px";
+			this.leftColumnWrapper.removeAttribute("border");
+			this.leftColumnWrapper.style.removeProperty("border-right");
+			this.leftColumnWrapper.setAttribute("class", "jumplinks-collapsed");
+		}
+		this.leftColumn.setAttribute("data-column-open", String(!previousState));
+		this.notch.setAttribute("data-column-open", String(!previousState));
+		this.notchIcon.setAttribute("data-column-open", String(!previousState));
+	}
+
+	render() {
+		const renderedHTML = this.traverse(this.filteredContents ?? this.content);
+		// listing the sidebar items on the basis of json
+		const sidebarMenuItems = Object.entries(this.content).map(
```

Comment:
have different files and functions for different sections, like for sidebar, search bar and filter